include/LightGBM/config.h

```diff
@@ -1134,7 +1134,7 @@ struct Config {
   static const std::string DumpAliases();
 
  private:
-  void CheckParamConflict();
+  void CheckParamConflict(const std::unordered_map<std::string, std::string>& params);
```

Comment:
`params` here contains the content of what was passed through by the used (with aliases already resolved). So it can be used to differentiate between "user code passed `num_leaves=31`" and "`num_leaves=31` because that's the default and user didn't pass any value for it".